yarn-project/circuits.js/src/structs/nullifier_key_validation_request.ts

```diff
@@ -29,7 +33,13 @@ export class NullifierKeyValidationRequest {
   }
 
   toFields(): Fr[] {
-    return [this.publicKey.toFields(), this.secretKey.high, this.secretKey.low].flat();
+    const fields = [this.publicKey.toFields(), this.secretKey.high, this.secretKey.low].flat();
+    if (fields.length !== NULLIFIER_KEY_VALIDATION_REQUEST_LENGTH) {
+      throw new Error(
+        `Invalid number of fields for NullifierKeyValidationRequest. Expected ${NULLIFIER_KEY_VALIDATION_REQUEST_LENGTH}, got ${fields.length}`,
+      );
+    }
+    return fields;
   }
 
   static fromFields(fields: Fr[] | FieldReader): NullifierKeyValidationRequest {
```

Comment:
<!--__GRAPHITE_HTML_TAG_START__--><i class='graphite__hidden'>[Re: line 62]<br/><br/></i>

<!--__GRAPHITE_HTML_TAG_END__-->Should this one be in its own file as well for consistency? `NullifierKeyValidationRequestContext`<!--__GRAPHITE_HTML_TAG_START__--><span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/4551?utm_source=unchanged-line-comment">Graphite</a>.</span>

<!--__GRAPHITE_HTML_TAG_END__-->

Reply:
Yes it should. I just felt like it should be done in a separate PR since this one is huge.